addons/website_slides/models/slide_channel.py

```diff
@@ -770,7 +792,24 @@ def _remove_membership(self, partner_ids):
         self.message_unsubscribe(partner_ids=partner_ids)
 
         if removed_channel_partner_domain:
-            self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain).unlink()
+            removed_channel_partner = self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain)
+            if removed_channel_partner:
+                removed_channel_partner.action_archive()
+
+    def _update_users_karma(self, partner_ids, joining=True):
+        """ Update the users karma while leaving / joining the course.
+
+            :param bool joining: If true, karma will added for the user(s),
+              otherwise (while leaving the course), it will be deducted.
+        """
+        earned_karma = self._get_earned_karma(partner_ids)
+        users = self.env['res.users'].sudo().search([
+            ('partner_id', 'in', list(earned_karma)),
+        ])
+        factor = 1 if joining else -1
+        for user in users:
+            if earned_karma[user.partner_id.id]:
```

Comment:
if it possible that `factor == 1` and `earned_karma[user.partner_id.id]` be not falsy ?
(it would means that you add existing karma, which is weird)

if not, I would move everything under `if not joining:`

Reply:
Hello @std-odoo 

I checked, and earned_karma is always Positive, no matter User is Joining Or Removing.

Thanks

Reply:
so you can apply my comment in't it ?

Reply:
Hello @std-odoo 

But here the case is different.
let's assume, the user earned the 20 Karma and left the course.
we are removing the karma from the user.

Then, the User joins the Course again and we are restoring the progress and Karma
so for that case we have to consider the factor.
if a user is removed from the course then -1 to remove it and if a user is joining the course again then we have to add existing karma.

Thanks

Reply:
ah ok I see now, thanks for the explanation :)

Reply:
Hi, 

Other changes are done.

can you please check it?

Thanks